src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -176,80 +176,95 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
         public void DirectoryInfo_Create_NullDirectoryInfo()
         {
             DirectoryInfo info = null;
-            DirectorySecurity security = new DirectorySecurity();
-
-            Assert.Throws<ArgumentNullException>("directoryInfo", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+            var security = new DirectorySecurity();
+            Assert.Throws<ArgumentNullException>("directoryInfo", () => DirectoryInfo_Create_Framework(info, security));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NullDirectorySecurity()
         {
-            DirectoryInfo info = new DirectoryInfo("path");
-
-            Assert.Throws<ArgumentNullException>("directorySecurity", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, null);
-                }
-                else
-                {
-                    info.Create(null);
-                }
-            });
+            var info = new DirectoryInfo("path");
+            Assert.Throws<ArgumentNullException>("directorySecurity", () => DirectoryInfo_Create_Framework(info, null));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NotFound()
         {
-            using var directory = new TempDirectory();
-            string path = Path.Combine(directory.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
-            DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+            var directory = new TempAclDirectory();
+            string path = Path.Combine(directory.Path, "ParentDoesNotExist");
+            directory.Dispose(); // Delete parent folder
 
-            Assert.Throws<UnauthorizedAccessException>(() =>
+            var info = new DirectoryInfo(path);
+            DirectorySecurity security = GetDirectorySecurity(FileSystemRights.FullControl);
+            DirectoryInfo_Create_Framework(info, security);
+        }
+
+        private void DirectoryInfo_Create_Framework(DirectoryInfo info, DirectorySecurity security)
```

Comment:
```suggestion
        private void CreateDirectoryInfoWithSecurity(DirectoryInfo info, DirectorySecurity security)
```
I don't see why use `Framework` on the name of this helper method.

Reply:
Would it be clearer if I renamed it to "DirectoryInfo_Create_SelectFramework"?

Reply:
OK, don't forget to rename similar method for `FileInfo`.

Reply:
I did.